src/main/java/org/jboss/netty/handler/ssl/SslHandler.java

```diff
@@ -1655,9 +1655,8 @@ public void run() {
 
                 if (futures != null) {
                     final ClosedChannelException cause = new ClosedChannelException();
-                    final int size = futures.size();
-                    for (int i = 0; i < size; i ++) {
-                        futures.get(i).setFailure(cause);
+                    for (ChannelFuture future : futures) {
```

Comment:
I changed that and removed ```size``` variable and left ```futures.size()```, the rest I kept the same. I force pushed so it is the same set of changes but no iterator.